src/controller/beast.ts

```diff
@@ -2,13 +2,100 @@ import { Request, Response } from 'express';
+     * @param req The request which contains the id of the requested beast
+     * @param res The response which will contain the requested beast
+     */
+    public static async getBeastById(req: Request, res: Response): Promise<void> {
+        let beast = null;
+        try {
+            const { id } = req.params;
+            if (!Number.isNaN(parseInt(id, 10))) {
+                beast = await BeastModel.getBeastById(parseInt(id, 10));
+            } else {
+                // Handle the case where id is not provided as a numerical string
+                // Need better error handling or sanitization here imo
+                res.status(400).json({ error: 'Invalid ID format' });
+            }
+            res.status(200).json({ message: 'Successfully retrieved', data: beast });
+        } catch (error: unknown) {
+            if (error instanceof CustomError) {
+                res.status(error.statusCode).json({ error: error.message });
+            }
+        }
+    }
+
+    /**
+     * Get a beast's friends by its id
+     * @param req the request which contains the id of the requested beast
+     * @param res the response which will contain the requested beast's friends
+     */
+    public static async getBeastFriends(req: Request, res: Response): Promise<void> {
+        let beast = null;
+        try {
+            const { id } = req.params;
+            if (!Number.isNaN(parseInt(id, 10))) {
+                beast = await BeastModel.getBeastFriends(parseInt(id, 10));
+            } else {
+                // Handle the case where id is not provided as a numerical string
+                // Need better error handling or sanitization here imo
+                res.status(400).json({ error: 'Invalid ID format' });
+            }
+            res.status(200).json({ message: 'Successfully retrieved', data: beast });
+        } catch (error: unknown) {
+            if (error instanceof CustomError) {
+                res.status(error.statusCode).json({ error: error.message });
+            }
+        }
+    }
+
+    /**
+     * Updates a beast with the given (optional) gamerTag and (optional) email
+     * @param req the request which contains the id of the beast to update, and the new gamerTag and email
+     * @param res the response which will contain a message indicating the success of the update
+     */
+    public static async updateBeast(req: Request, res: Response): Promise<void> {
+        try {
+            const { gamerTag, email } = req.body;
+            const { id } = req.params;
+
+            if (Number.isNaN(parseInt(id, 10))) {
+                // Handle the case where id is not provided as a numerical string
+                // Need better error handling or sanitization here imo
+                res.status(400).json({ error: 'Invalid ID format' });
+            }
+
+            if (!gamerTag && !email) {
+                res.status(400).json({ error: 'Must provide at least one updated parameter' });
```

Comment:
Do we think this error is necessary? I think it's unlikely this request will ever contain an empty body. If you do want to keep it, I'd check the req.body rather than each individual field since it can scale to future update fields

Reply:
Removed

Reply:
I think this error should still be handled but in the validator middleware

Reply:
Yea I agree , it is gone as of now however